crates/rust-analyzer/build.rs

```diff
@@ -5,11 +5,13 @@ use std::{env, path::PathBuf, process::Command};
 fn main() {
     set_rerun();
 
-    let rev = rev().unwrap_or_else(|| "???????".to_string());
+    let rev = env::var("REV").ok().or(rev()).unwrap_or_else(|| "???????".to_string());
```

Comment:
```suggestion
    let rev = env::var("RUST_ANALYZER_REV").ok().or_else(rev).unwrap_or_else(|| "???????".to_string());
```

Other crates may do something similar to get the revision name.

Reply:
Doesn't them share the same revision?

Reply:
This is the rust analyzer revision, which uses either git commit hash or now a release date, other crates may use a different revision system. For example they may want to add a release channel (rustc) or something else.